ci/dependency-update/Jenkinsfile

```diff
@@ -113,18 +117,45 @@ pipeline {
 			post {
 				cleanup {
 					sh 'ci/docker-cleanup.sh'
+					dir('hibernate-orm-local-copy') {
+						deleteDir()
+					}
 				}
 			}
-			steps {
-				// The timeout cannot be in stage options, because that would
-				// include the time needed to provision a node.
-				timeout(time: 30, unit: 'MINUTES') {
-					withMavenWorkspace {
-						sh """ \
+			stages {
+				stage('Build Hibernate ORM') {
+					when {
+						expression {
+							return params.ORM_REPOSITORY?.trim() || params.ORM_BRANCH?.trim()
+						}
+					}
```

Comment:
It seems this expression is evaluated after allocating a node to the stage, which will be a waste if you split this stage from the "Build Hibernate Search" stage as I recommended in another comment.

Adding `beforeAgent true` should fix it; see https://www.jenkins.io/doc/book/pipeline/syntax/#evaluating-when-before-entering-agent-in-a-stage